landscaper/pkg/gardenlet/chart/charttest/charttest.go

```diff
@@ -825,14 +825,21 @@ func ComputeExpectedGardenletConfiguration(
 }
 
 // VerifyGardenletComponentConfigConfigMap verifies that the actual Gardenlet component config config map equals the expected config map.
-func VerifyGardenletComponentConfigConfigMap(ctx context.Context, c client.Client, universalDecoder runtime.Decoder, expectedGardenletConfig gardenletconfigv1alpha1.GardenletConfiguration, expectedLabels map[string]string) {
+func VerifyGardenletComponentConfigConfigMap(
```

Comment:
This test is about to be removed with https://github.com/gardener/gardener/pull/5481. If we want to keep the test, we have to rather move it to ManagedSeed actuator tests (currently unit tests of this component use a mock Client/ChartApplier and only verify that the `Apply()` was called with the expected args).
Asking myself whether the unit test would be useful and would be able to catch a potential regression - I think yes, it should be able to catch for example formatting errors such as https://github.com/gardener/gardener/pull/5223. On the other side create ManagedSeed integration test should also be able to catch similar formatting errors or other errors that prevent the chart to be applied properly. If it is getting too much effort to maintain it, then I am also fine to drop it.
At the end I think it boils down to the question on how we test charts - do we want to have unit tests for components that apply charts or do we rather rely on e2e/integration tests for this.

Reply:
I agree with your point. My points:

Pro using integration tests for chart testing:
- There are a lot of test cases in `gardenlet_chart_test.go`. A simple change in the default gardenlet configuration will need a lot of adaption in the tests now with the unique names. This is cumbersome. Integration tests don't need that but test it implicitly.

Pro unit tests:

- the unit tests capture the complexity of our configuration by creating a new `Entry` in the testTable for each configuration option. The integration tests would probably only test a subset of the configuration paths.
- Spotting a formatting error early in the unit test phase saves time in comparison to catching it only in the integration tests.

All in all I suggest to keep those tests for now and move them to the manageedSeed Actuator tests. 
Any other opinions?

Reply:
@danielfoehrKn Would this be fine for you?